cmd/authd-oidc/daemon/daemon.go

```diff
@@ -142,20 +149,18 @@ func (a *App) serve(config daemonConfig) error {
 
 	s, err := dbusservice.New(ctx, b)
 	if err != nil {
-		close(a.ready)
 		return err
 	}
 
 	var daemonopts []daemon.Option
 	daemon, err := daemon.New(ctx, s, daemonopts...)
 	if err != nil {
 		_ = s.Stop()
-		close(a.ready)
 		return err
 	}
 
 	a.daemon = daemon
-	close(a.ready)
+	closeFunc()
```

Comment:
We could just not call it here and let the deferred function call run on return, right? Then we don't even need the trick with checking the pointer and setting it to nil, because the function is always run exactly once.

Reply:
Yeah, I was thinking about but I didn't want to change the behavior that the `close` happens *before* having finished serving, while by doing that it would happen after.